source/common/ssl/context_impl.h

```diff
@@ -138,9 +146,17 @@ class ServerContextImpl : public ContextImpl, public ServerContext {
 private:
   int alpnSelectCallback(const unsigned char** out, unsigned char* outlen, const unsigned char* in,
                          unsigned int inlen);
+  int sessionTicketProcess(SSL* ssl, uint8_t* key_name, uint8_t* iv, EVP_CIPHER_CTX* ctx,
+                           HMAC_CTX* hmac_ctx, int encrypt);
 
   Runtime::Loader& runtime_;
   std::vector<uint8_t> parsed_alt_alpn_protocols_;
+  struct SessionTicketKey {
+    std::array<uint8_t, SSL_TICKET_KEY_NAME_LEN> name;
+    std::array<uint8_t, 32> aes_key;
```

Comment:
AES key should be after HMAC key, this way keying material will be compatible with NGINX and possibly other proxies.